core/src/main/java/org/commonjava/propulsor/boot/Booter.java

```diff
@@ -123,120 +118,114 @@ private static BootOptions loadBootOptions()
-            }
-
+            String home = System.getenv( options.getHomeEnvar() );
             if ( home == null )
             {
-                home = new File( "." ).getCanonicalPath();
+                home = System.getProperty( options.getHomeSystemProperty(), new File( "." ).getCanonicalPath() );
             }
+            options.setHomeDir( home );
 
-            options.load( bootDefaults, home );
+            options.load( bootDefaults );
             return options;
         }
-        catch ( final IOException e )
+        catch ( final Exception e )
         {
             throw new BootException( "ERROR LOADING BOOT DEFAULTS: %s.\nReason: %s\n\n", e, bootDefaults,
                                      e.getMessage() );
         }
-        catch ( final InterpolationException e )
-        {
-            throw new BootException( "ERROR RESOLVING BOOT DEFAULTS: %s.\nReason: %s\n\n", e, bootDefaults,
-                                     e.getMessage() );
-        }
+    }
+
+    public static void setUncaughtExceptionHandler()
+    {
+        Thread.currentThread().setUncaughtExceptionHandler( ( thread, error ) -> {
+            if ( error instanceof InvocationTargetException )
+            {
+                final InvocationTargetException ite = (InvocationTargetException) error;
+                System.err.println( "In: " + thread.getName() + "(" + thread.getId()
+                                                    + "), caught InvocationTargetException:" );
+                ite.getTargetException().printStackTrace();
+
+                System.err.println( "...via:" );
+                error.printStackTrace();
+            }
+            else
+            {
+                System.err.println( "In: " + thread.getName() + "(" + thread.getId() + ") Uncaught error:" );
+                error.printStackTrace();
+            }
+        } );
     }
 
     private final Logger logger = LoggerFactory.getLogger( getClass() );
 
-    private BootStatus status;
+    protected BootStatus status;
 
-    private BootOptions options;
+    protected BootOptions options;
 
-    private Weld weld;
+    protected Weld weld;
 
-    private WeldContainer container;
+    protected WeldContainer container;
 
-    private Deployer deployer;
+    protected Deployer deployer;
 
-    private Configurator configurator;
+    protected Configurator configurator;
 
-    private AppLifecycleManager lifecycleManager;
+    protected AppLifecycleManager lifecycleManager;
 
-    public BootStatus initialize( final BootOptions options )
-            throws BootException
+    @Override
+    public void initialize( final BootOptions options ) throws BootException
     {
         this.options = options;
+        this.options.setSystemProperties();
 
-        try
-        {
-            options.setSystemProperties();
+        weld = new Weld();
+        weld.property( "org.jboss.weld.se.archive.isolation", false );
```

Comment:
Does this mean we don't really need to create things like the indy embedder anymore, finally?